lib/tasks/pubmed.rake

```diff
@@ -10,4 +10,36 @@ namespace :pubmed do
     doc = client.fetch_records_for_pmid_list(pmids)
     puts doc # XML document
   end
+
+  desc 'Harvest using a text file with a list of publications by PubmedID for the supplied cap_profile_id'
+  task :pmid_profile_id_import, [:path_to_report,:cap_profile_id] => :environment do |_t, args|
+    lines = IO.readlines args[:path_to_report]
+    author = Author.find_by_cap_profile_id(args[:cap_profile_id])
+
+    puts "Cap_profile_id #{args[:cap_profile_id]} is #{author.first_name} #{author.last_name}"
+    total_pub_count=author.contributions.size
+    new_pub_count=author.contributions.where(:status=>'new').size
+    approved_pub_count=author.contributions.where(:status=>'approved').size
+    puts "total publications: #{total_pub_count}"
+    puts "total new publications: #{new_pub_count}"
+    puts "total accepted publications: #{approved_pub_count}"
+    puts "attempting to harvest #{lines.count} new publications by pubmedID"
+    puts
+
+    harvester=ScienceWireHarvester.new
+    lines.each do |line|
+      pmid=line.chomp
+      puts "working on #{pmid}..."
+      pub=Publication.find_or_create_by_pmid(pmid)
+      harvester.add_contribution_for_harvest_suggestion(author,pub)
+      pub.add_all_db_contributions_to_my_pub_hash
+      pub.save
+    end
+
+    author.reload
+    new_pub_count=author.contributions.where(:status=>'new').size - new_pub_count
+    puts ""
+    puts "added publications: #{new_pub_count}"
+
```

Comment:
:+1: - looks like good feedback to the console on how things are going.  But, wondering what happens when there is an exception raised for any one of the `line` values?  It might be worth using a `begin - rescue` block within the `line` iteration to catch occasional exceptions and continue on with the next line.